BeagleBone_black/src/communications/main.cpp

```diff
@@ -138,6 +138,9 @@ int Main::sendTorqueBl(float torquebl)
 
 void Main::run()
 {
+  data::Communications cmn_data;
+  cmn_data.module_status = data::ModuleStatus::kInit;
```

Comment:
How do you know at this point that you have everything initialised correctly (e.g. the comn socket is open?)

Reply:
I do not think this method would run if the socket did not establish a connection. That may be some confusion on my part though. I can move this to right after connection is established in baseCommunicator.cpp to make sure.